### ISSUE TITLE
Docs: Add user facing docs for running a separate Task Execution API-server

### ISSUE THREAD

**Issue by kaxil** [apache/airflow#43103]
In https://github.com/apache/airflow/pull/43015, a new Task Execution API-server was added. This can be run with the Core Airflow APIs and can also be run separately.

To run all apis
```shell
airflow api-server

# Alternate
airflow api-server --apps all

# Alternate
airflow api-server--apps core,execution
```

E.g to only run Task execution API
```shell
airflow api-server --apps execution
```

The doc should contain details of our recommendation.

**Comment by Brijeshthummar02** [comment]
@kaxil i can work on it, could you show me the path to files, and where to add it also assign me.

**Comment by kaxil** [comment]
Hey @Brijeshthummar02 , [AIP-72](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-72+Task+Execution+Interface+aka+Task+SDK?src=contextnavpagetreemode) is still under progress: https://github.com/orgs/apache/projects/405

This IMO would be need some background about [AIP-72](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-72+Task+Execution+Interface+aka+Task+SDK?src=contextnavpagetreemode), so unless you have read it might be difficult.

Keeping it for myself or @ashb to work on it once we are done with that AIP.

**Comment by jason810496** [comment]
Self assign myself, will raise PR after #52929 merge, just added to my backlog.
